### PR TITLE
Use allowed-list of routes to pass through to JS route-handler instead of catch-all.

### PR DESCRIPTION
**Pull request by jgerigmeyer**
## Cute animal pic
_Because everyone likes pictures of [animals](https://unsplash.com/t/animals)._

penguins

## Link to Trello card (if applicable)
_If this PR relates to a Trello card, don't forget to reference this PR on the card as well._

https://trello.com/c/DU4Na7b7

## Description
_The commit messages say what you did; this should explain why and/or how._
- [x] Description is in Trello card

### CODE REVIEW

**Comment by jgerigmeyer** [comment]
[Remove catch-all route.](https://trello.com/c/DU4Na7b7/203-remove-catch-all-route)

**Comment by jgerigmeyer** [comment]
> I am 👍 on this, but I want to ask who's gonna be "owner" of it (I hate that term) for making sure it's kept up to date with whatever the frontend paths are. I'm happy to, but it just seemed a question important to ask.

Yeah, that's a fair question. I'm guessing the person adding a new front-end route would "own" this by default, since it really won't work until this is updated accordingly. 🤷

**Comment by jgerigmeyer** [comment]
@wlonk Requested re-review since I added https://github.com/oddbird/Metecho/pull/330/commits/5a301b4474771b8d3e97d8e8cb3cf4f0a72c233f

File: metecho/urls.py [hunk 1]
```diff
@@ -36,11 +36,9 @@
     path(PREFIX + "/", admin.site.urls),
     path("accounts/", include("allauth.urls")),
     path("api/", include("metecho.api.urls")),
-    # Catchall for the rest. Right now, it just trusts that PREFIX ==
-    # 'admin', because we don't want to do string munging to get just
-    # the part without the regex and path cruft on it.
+    # Routes to pass through to the front end JS route-handler
     re_path(
-        r"^(?!{admin}|accounts|api|static)".format(admin=PREFIX),
+        r"^($|login\/?$|terms\/?$|repositories(\/|$)|accounts(\/|$))",
```
Comment (jgerigmeyer) [inline metecho/urls.py:41]: I, uh, am not real confident when it comes to writing a regex... 😬

Comment (wlonk) [reply metecho/urls.py:41]: Ah, that's better, from a pedantic perspective :D

Comment (jgerigmeyer) [reply metecho/urls.py:41]: That's my goal! 🎉

**Pull request merged**